File: _project_review/proyecto-final-front-errorcapa8-dev/backend/src/main/java/co/icesi/TallerJPA/model/Space.java
package co.icesi.TallerJPA.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import java.time.LocalDateTime;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "spaces")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Space {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String type;
    private String description;
    private String location;
    private boolean available;
    private int capacity;
    private String openingHours;
    private String closingHours;
    private boolean active = true;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToMany(mappedBy = "spaces")
    private List<Event> events;

    @PrePersist
    void prePersist() {
        LocalDateTime now = LocalDateTime.now();
        createdAt = now;
        updatedAt = now;
    }

    @PreUpdate
    void preUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: _project_review/proyecto-final-front-errorcapa8-dev/front/bienestar-app/src/types/space.ts
export interface Space {
  id: number;
  name: string;
  type: string;
  description: string;
  location: string;
  available: boolean;
  capacity: number;
  openingHours?: string | null;
  closingHours?: string | null;
  active: boolean;
}

export interface SpaceRequest {
  name: string;
  type: string;
  description?: string;
  location?: string;
  available: boolean;
  capacity: number;
  openingHours?: string;
  closingHours?: string;
}
